src/deploy/functions/params.ts

```diff
@@ -444,6 +497,52 @@ async function promptParam(
   assertExhaustive(param);
 }
 
+async function promptList(param: ListParam, resolvedDefault?: string[]): Promise<string[]> {
+  if (!param.input) {
+    const defaultToText: TextInput<string> = { text: {} };
+    param.input = defaultToText;
+  }
+  let prompt: string;
+  const stringToList = (res: string): string[] | retryInput => {
+    const converted = JSON.parse(res) || [];
+    if (!Array.isArray(converted)) {
+      return { message: `"${res}" is not an array` };
+    }
+    for (const entry of converted) {
+      if (typeof entry !== "string") {
+        return { message: `Parsed array contains a non-string element ${entry}` };
+      }
+    }
+    return converted as string[];
+  };
+
+  if (isSelectInput(param.input)) {
+    throw new FirebaseError("List params cannot have non-list selector inputs");
+  } else if (isListSelectInput(param.input)) {
+    prompt = `Select a value for ${param.label || param.name}:`;
+    if (param.description) {
+      prompt += ` \n(${param.description})`;
+    }
+    prompt += "\nSelect an option with the arrow keys, and use Enter to confirm your choice. ";
+    return promptSelectMultiple<string>(
+      prompt,
+      param.input,
+      resolvedDefault,
+      (res: string[]) => res
+    );
+  } else if (isTextInput(param.input)) {
+    prompt = `Enter a list of strings for ${param.label || param.name}:`;
```

Comment:
This will probably get more complex if/when we transition to inputting with a delimiter. If your delimiter is newline, you probably have to accept double newlines as the way you end your prompt. IDK if the prompt library supports that, so we _might_, _for now_ stick to just comma delimiters.